src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs

```diff
@@ -72,5 +77,52 @@ protected IList<TaggedText> GetAwaitableUsage(IMethodSymbol method, SemanticMode
 
             return SpecializedCollections.EmptyList<TaggedText>();
         }
+
+        protected ISymbol GuessCurrentSymbol(SeparatedSyntaxList<ArgumentSyntax> arguments, ImmutableArray<IMethodSymbol> methodGroup,
+            SemanticModel semanticModel, ISemanticFactsService semanticFactsService, CancellationToken cancellationToken)
+        {
+            return methodGroup.FirstOrDefault(m => isAcceptable(m));
+
+            bool isAcceptable(IMethodSymbol method)
+            {
+                if (arguments.Count == 0)
+                {
+                    return false;
+                }
```

Comment:
1. 'IsAcceptable' :)  This is hte IDE.
2. If yo haven't no arguments, aren't *all* methods acceptable?

Reply:
1. Sorry about that. Habits
2. If the provider recommends a selection in such case, then if you type an argument then delete it, the selection will be reset to the first in the list (which is not very useful). If the provider makes no recommendation in this case, the model sticks with the last recommendation. I'll add a comment.